cdap-security/src/main/java/co/cask/cdap/security/guice/SecureStoreModules.java

```diff
@@ -50,70 +53,65 @@
-      @Override
-      protected void configure() {
-        bind(SecureStore.class)
-          .annotatedWith(Names.named(DELEGATE_SECURE_STORE))
-          .toProvider(new TypeLiteral<StoreProvider<SecureStore>>() { });
-        bind(SecureStoreManager.class)
-          .annotatedWith(Names.named(DELEGATE_SECURE_STORE_MANAGER))
-          .toProvider(new TypeLiteral<StoreProvider<SecureStoreManager>>() { });
-        bind(SecureStore.class).to(DefaultSecureStoreService.class);
-        bind(SecureStoreManager.class).to(DefaultSecureStoreService.class);
-        expose(SecureStore.class);
-        expose(SecureStoreManager.class);
-      }
-    };
+    return new LocalModule();
   }
 
   @Override
   public final Module getStandaloneModules() {
-    return new PrivateModule() {
-      @Override
-      protected void configure() {
-        bind(SecureStore.class)
-          .annotatedWith(Names.named(DELEGATE_SECURE_STORE))
-          .toProvider(new TypeLiteral<StoreProvider<SecureStore>>() { });
-        bind(SecureStoreManager.class)
-          .annotatedWith(Names.named(DELEGATE_SECURE_STORE_MANAGER))
-          .toProvider(new TypeLiteral<StoreProvider<SecureStoreManager>>() { });
-        bind(SecureStore.class).to(DefaultSecureStoreService.class);
-        expose(SecureStore.class);
-        bind(SecureStoreManager.class).to(DefaultSecureStoreService.class);
-        expose(SecureStoreManager.class);
-      }
-    };
+    return new LocalModule();
+  }
+
+  /**
+   * Guice module being used in in memory as well as standalone mode.
+   */
+  private final class LocalModule extends PrivateModule {
+
+    @Override
+    protected void configure() {
+      bind(SecureStoreService.class)
+        .annotatedWith(Names.named(DELEGATE_SECURE_STORE_SERVICE))
+        .toProvider(StoreProvider.class);
+
+      bind(DefaultSecureStoreService.class).in(Scopes.SINGLETON);
+      bind(SecureStore.class).to(DefaultSecureStoreService.class);
+      bind(SecureStoreManager.class).to(DefaultSecureStoreService.class);
+      bind(SecureStoreService.class).to(DefaultSecureStoreService.class);
+      expose(SecureStore.class);
+      expose(SecureStoreManager.class);
+      expose(SecureStoreService.class);
+    }
   }
 
   @Override
   public final Module getDistributedModules() {
     return new PrivateModule() {
       @Override
       protected void configure() {
-        bind(SecureStore.class)
-          .annotatedWith(Names.named(DELEGATE_SECURE_STORE))
-          .toProvider(new TypeLiteral<DistributedStoreProvider<SecureStore>>() { });
-        bind(SecureStoreManager.class)
-          .annotatedWith(Names.named(DELEGATE_SECURE_STORE_MANAGER))
-          .toProvider(new TypeLiteral<DistributedStoreProvider<SecureStoreManager>>() { });
+        bind(SecureStoreService.class)
+          .annotatedWith(Names.named(DELEGATE_SECURE_STORE_SERVICE))
+          .toProvider(new TypeLiteral<DistributedStoreProvider<SecureStoreService>>() { });
```

Comment:
Do we still need two different providers? Seems like they are doing very similar job, which both of them try to provide a `SecureStoreService` based on configuration.